s3sync.go

```diff
@@ -144,6 +144,62 @@ func (m *Manager) Sync(source, dest string) error {
 	return errors.New("local to local sync is not supported")
 }
 
+func (m *Manager) SyncWithCtx(ctx context.Context, source, dest string) error {
```

Comment:
I think `SyncWithContext` is more common in Go official libraries
e.g. `http.NewRequestWithContext`